### ISSUE TITLE
Feature request: include an actual command line interface to dngrep

### ISSUE THREAD

**Issue by enthusedcoder** [dnGrep/dnGrep#1015]
So I recently started playing around with dngrep's scripting feature, and I got to say, I really like the ability to create these automation scripts to automate simple search and replace actions rather than having to plug everything into the gui, especially with my highly repetitive tasks.  However, there is one criticism that I have of the feature, and that is the lack of an actual command line interface, as, when the scripts are run, they simply automate the configuration and execution of the search/replace via the program's main gui.  This has several consequences that make using the scripting feature as part of an automated solution more complicated:

1. (and probably most importantly) The command line immediately returns upon execution of the application, but does not wait for dngrep to perform the actions that are defined in the script.  This means that, if I am trying to incorporate a dngrep seach/replace using a script into an automation of some kind, the automation will immediately move on to the next step before dngrep has acutally done what it needs to, and if your next step is dependent upon the actions of the previous dngrep script having already taken place, well, you are SOL.  The only way around this would be to configure the automation to pause execution of the automation for a set period to wait for the dngrep script to execute, but that is not at all efficient and can result in errors.
2. Cannot run automation scripts remotely: As you are aware, when a script is run from the command line, the first thing that happens is the application GUI is launched, as all the scripts do is automate the GUI.  However, this means that it would be impossible to run these automation scripts remotely using tools such as psexec or powershell remoting **unless** the user which you are using to login to the remote computer is already actively logged on (to allow the GUI to launch).  AGain, not ideal.

Including an actual command line interface for dngrep would resolve both of these issues immediately, so I do hope that you will include one in a future release of dngrep.  Thanks!!

**Comment by doug24** [comment]
Yes, the scripting is just automating the GUI application. There was a recent issue about running the app without the UI from the command line, and I did look at it some at that time.  I concluded it would take a lot of work to change it to run without the UI.  I will think about it some more, but no promises.

**Comment by wdlq** [comment]
> Yes, the scripting is just automating the GUI application. There was a recent issue about running the app without the UI from the command line, and I did look at it some at that time. I concluded it would take a lot of work to change it to run without the UI. I will think about it some more, but no promises.

Some basic functions  are enough. For example, run the command line "dnGrep.exe -f C:\folder -s Keyword -pm *.txt" to return a list of  txt file paths which contains the Keyword.

**Comment by wdlq** [comment]
I had implemented this function by refer to Everything64.dll in my code.  Thank you for your source code.

**Comment by doug24** [comment]
The thing that makes this a lot of work is to be able to run the application without opening the main window of the app or showing any message boxes.  It doesn't matter if the command line is simple or not.  

Have you tried looking at [ripgrep](https://github.com/BurntSushi/ripgrep) for a command line tool?

**Comment by doug24** [comment]
This would require a major rewrite of the application.  I'm not going to take it on, and no one else has expressed interest in it, so I'm closing this issue.
